cleverhans/attacks_tf.py

```diff
@@ -16,31 +16,55 @@
 FLAGS = flags.FLAGS
 
 
-def fgsm(x, predictions, eps, clip_min=None, clip_max=None):
+def fgsm(x, predictions, eps=0.3, clip_min=None, clip_max=None):
+    warnings.warn("attacks_tf.fgsm is deprecated and will be removed on "
+                  " 09-27-17. Use attacks_tf.fgm instead.")
+    return fgm(x, predictions, y=None, eps=eps, ord=np.inf, clip_min=clip_min,
+               clip_max=clip_max)
+
+
+def fgm(x, preds, y=None, eps=0.3, ord=np.inf, clip_min=None, clip_max=None):
     """
-    TensorFlow implementation of the Fast Gradient
-    Sign method.
+    TensorFlow implementation of the Fast Gradient Method.
     :param x: the input placeholder
-    :param predictions: the model's output tensor
+    :param preds: the model's output tensor
+    :param y: the output placeholder. Use None (default) to avoid label leaking
     :param eps: the epsilon (input variation parameter)
-    :param clip_min: optional parameter that can be used to set a minimum
-                    value for components of the example returned
-    :param clip_max: optional parameter that can be used to set a maximum
-                    value for components of the example returned
+    :param ord: string with the norm order to use when computing gradients.
+                This should be either np.inf, 1 or 2.
+    :param clip_min: Minimum float value for adversarial example components
+    :param clip_max: Maximum float value for adversarial example components
     :return: a tensor for the adversarial example
     """
 
-    # Compute loss
-    y = tf.to_float(
-        tf.equal(predictions, tf.reduce_max(predictions, 1, keep_dims=True)))
+    if y is None:
+        # Using model predictions as ground truth to avoid label leaking
```

Comment:
I'd put "label leaking" in quotes and put https://arxiv.org/abs/1611.01236 on the next line, or most users won't know what this comment means